src/pages/auth/signup.js

```diff
@@ -0,0 +1,192 @@
+import { useState, useEffect, useRef } from "react";
+import axios from "axios";
+import Router from "next/router";
+
+import Head from "next/head";
+import Link from "next/link";
+import { makeStyles } from "@material-ui/core/styles";
+import { Container, TextField, Typography, Button } from "@material-ui/core";
+import TransitionAlert from "../../components/TransitionAlert/TransitionAlert";
+
+const useStyles = makeStyles((theme) => ({
+    paper: {
+        marginTop: theme.spacing(10),
+        display: "flex",
+        flexDirection: "column",
+        alignItems: "center",
+    },
+    form: {
+        "& > *": {
+            margin: theme.spacing(1),
+            width: "40ch",
+        },
+        display: "flex",
+        flexWrap: "wrap",
+        flexDirection: "column",
+        alignItems: "center",
+        marginTop: theme.spacing(1),
+    },
+    link: {
+        textDecoration: "none",
+        color: theme.palette.info.main,
+    },
+    errorMessage: {
+        color: theme.palette.error.main,
+        margin: 0,
+    },
+}));
+
+export default function Login() {
+    const classes = useStyles();
+    const isInitialMount = useRef(true);
+
+    const [email, setEmail] = useState("");
+    const [isEmailValid, setIsEmailValid] = useState(true);
+    const [password, setPassword] = useState("");
+    const [confirmPassword, setConfirmPassword] = useState("");
+    const [isPasswordMatch, setIsPasswordMatch] = useState(true);
+
+    const [isOpenAlert, setIsOpenAlert] = useState(false);
+
+    const handleSubmit = (event) => {
+        event.preventDefault();
+        const payload = {
+            email,
+            password,
+            confirmPassword,
+        };
+
+        axios
+            .post("http://localhost:3000/api/signup", payload)
+            .then((res) => Router.push("/"))
+            .catch((err) => {
+                setIsOpenAlert(true);
+                console.error(err);
+            });
+    };
+
+    // email validation
+    useEffect(() => {
+        // only validate when email change and not on initial mount
+        if (isInitialMount.current) {
+            isInitialMount.current = false;
+        } else {
+            const re =
+                /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+
+            const isValid = re.test(String(email).toLowerCase());
+            setIsEmailValid(isValid);
+        }
+    }, [email]);
+
+    // validate password
+    useEffect(() => {
+        // only validate when password change and not on initial mount
+        if (isInitialMount.current) {
+            isInitialMount.current = false;
```

Comment:
Asking, so for the initial mount, it will not check the password yaa? but how about we just check the password and confirmPassword is exist, so we can remove isInitialMount variable?

Wdyt if like this?
```
if (password && confirmPassword) {
    if (password === confirmPassword) {
        ...
    } else {
        ...
    }
}
```

Reply:
it could do, but i think the error validation message would show up before user typing anything yet because the field is empty

Reply:
I see it's also good concern then 👍 

But my point is how to reduce our variables, previously I thinking to show the error if the password and confirm password is not same, and because your initial state is `true`, I think the message would not show up before user type anything
```
 {password && confirmPassword && !isPasswordMatch && (
    <label className={classes.errorMessage}>
        {"Password didn't match"}
    </label>
 )}
```

Reply:
ahh i see, will change it kak

Reply:
If done, you can merge yaa, for now is good 👍